.github/workflows/on-pullrequest.yml

```diff
@@ -27,12 +27,12 @@ concurrency:
   group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
   cancel-in-progress: true
 
-permissions:
-  checks: write
-  pull-requests: write
-  actions: read
-  contents: read
-  security-events: write
+permissions: read-all
+  #checks: write
+  #pull-requests: write
+  #actions: read
+  #contents: read
+  #security-events: write
```

Comment:
You'll need to put `security-events: write` explicitly into the `code-analysis` job.

Once you do that, in your fork, can you merge your changes to main and then create a new pull request to test that the workflow runs properly?

Reply:
@stephenfuqua I've made changes to the file permissions. I attempted to test other workflows by creating a Release/Pre-release, but encountered the errors mentioned below.

**- After Pull request:**   _Done_
**- On Merge to Main or Releasable Tag:**     _{   "message": "Bad credentials",   "documentation_url":_
**- On Pre-Release:**    _Failure on Create NuGet Package_
**- On Pull Request:**    _Done_
**- On Pull Request Dockerfile:**    _Username and password required_
**- On Release:**    _Promote Azure Artifact error_